### ISSUE TITLE
[2.0.x]Compilation error "Error: incorrect register `%edi' used with `q' suffix"

### ISSUE THREAD

**Issue by johngray** [phalcon/cphalcon#10427]
Hi. I am trying to compile the latest phalcon release. I am getting this [error (gist)] (https://gist.github.com/johngray/4533aa3fdc46bf2f1bee)
Thus I cannot build phalcon. The build directory doesn't matter (cphalcon/build or cphalcon/ext) 

cpuid:
```
Vendor ID: "GenuineIntel"; CPUID level 13
Extended brand string: "       Intel(R) Core(TM) i7-3520M CPU @ 2.90GHz"
```
gcc --version:
```
gcc (Ubuntu/Linaro 4.7.2-2ubuntu1) 4.7.2
```
uname -a:
```
Linux p 3.5.0-17-generic #28-Ubuntu SMP Tue Oct 9 19:31:23 UTC 2012 x86_64 x86_64 x86_64 GNU/Linux
```

Thanks for any help.

**Comment by Green-Cat** [comment]
Are you compiling master or the 2.0.x branch?

**Comment by johngray** [comment]
Master
git status output:
```
# On branch master
nothing to commit (working directory clean)
```

**Comment by johngray** [comment]
Compiling 2.0.x  and 2.0.0 gives the same error

**Comment by andresgutierrez** [comment]
Any ideas how to reproduce this?

**Comment by johngray** [comment]
Nope. I am not very skilled in C. But I suspect the wrong usage of registers.
I found a similar thread on the [forum](http://forum.phalconphp.com/discussion/6457/error-compilation-phalcon-20)
And also I found an old thread with similar error on [gcc bugzilla](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=13167)
Maybe this will provide you with some clue.

**Comment by andresgutierrez** [comment]
I found this was a bug in PHP and it was  fixed in PHP 5.5.0beta4 https://github.com/phalcon/cphalcon/issues/10501

**Comment by johngray** [comment]
Thanks. I upgraded to 5.6.10 and now everything is fine.
